Libraries/src/net/relinc/libraries/sample/LoadDisplacementSampleResults.java

```diff
@@ -363,4 +405,80 @@ public double getNumberOfReflections(){
 		return SPOperations.getDerivative(time, displacement);
 	}
 
+	public DataSubset getCurrentDisplacementDatasubset(){
+		return sample.getDataSubsetAtLocation(displacementDataLocation);
+	}
+
+	public DataLocation getCurrentDisplacementLocation()
+	{
+		return displacementDataLocation;
+	}
+
+	public DataSubset getCurrentLoadDatasubset(){
+		return sample.getDataSubsetAtLocation(loadDataLocation);
+	}
+
+	public DataLocation getCurrentLoadLocation()
+	{
+		return loadDataLocation;
+	}
+
+	public boolean isFaceForceGraphable(){
+		return getCurrentLoadDatasubset() instanceof TransmissionPulse
+				&& getCurrentDisplacementDatasubset() instanceof ReflectedPulse
+				&& !(sample instanceof TorsionSample)
+				&& !(sample instanceof BrazilianTensileSample);
+	}
+
+	public double[] getFrontFaceForce()
+	{
+		ReflectedPulse reflectedPulse = (ReflectedPulse)getCurrentDisplacementDatasubset();
+		// TODO: this is assumes sample is HopkinsonBarSample...
+		Map<String, double[]> forceData = ((HopkinsonBarSample)sample).getFrontFaceForceInterpolated(reflectedPulse);
```

Comment:
meh, this would ideally be a class. if you are creating a map/dict and you know all the keys beforehand, it should be a class/struct imo. Gives the compiler more to work with. it is less code though.